### PR TITLE
fix: resolve flaky test issue by making MarkerSet preserve order

### PR DESCRIPTION
**Pull request by sivchari**
Problem:
- Multiple markers test was flaky, sometimes generating validators in wrong order
- MarkerSet was map-based, which has non-deterministic iteration order in Go

Solution:
- Changed MarkerSet from map[string]Marker to []Marker (slice-based)
- This preserves the definition order from source code
- No performance impact due to small number of markers per field

Changes:
- MarkerSet is now a slice type that preserves insertion order
- Added Add() method to handle duplicate checking
- Updated insertFieldMarker to work with single markers
- Simplified makeValidator since order is now guaranteed

This ensures consistent code generation and eliminates test flakiness.

## Type of Change

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation update
- [ ] Refactoring (no functional changes)

## Quality Checklist

Before submitting this pull request, ensure your contribution meets these quality standards:

### Code Quality
- [ ] Code passes `make golangci-lint` without errors
- [ ] All tests pass: `make test`
- [ ] Code follows existing patterns and conventions
- [ ] No hard-coded values (use constants or configuration)
- [ ] Error messages are clear and actionable

### Testing Requirements
- [ ] Golden tests created and passing (`internal/analyzers/govalid/testdata/`)
- [ ] Unit tests implemented with boundary value testing (`test/unit/`)
- [ ] Benchmark tests comparing against popular validation libraries (`test/benchmark/`)
  - [ ] [go-playground/validator](https://github.com/go-playground/validator) (BenchmarkGoPlayground*)
  - [ ] [asaskevich/govalidator](https://github.com/asaskevich/govalidator) (BenchmarkGoValidator*)
  - [ ] [gookit/validate](https://github.com/gookit/validate) (BenchmarkGookitValidate*)
- [ ] All test files generated: `cd test && go generate`

### Performance Standards
- [ ] Zero allocations in validation logic (verified via benchmarks)
- [ ] Performance improvement over existing validation libraries (minimum 2x faster)
- [ ] Benchmark results added to `test/benchmark/README.md`

### Documentation
- [ ] README.md updated with new marker documentation
- [ ] Code includes appropriate comments and examples
- [ ] Validator usage examples provided

### Integration
- [ ] Validator scaffold generated with `make generate-validator MARKER=yourmarker`
- [ ] Registry files automatically updated
- [ ] Binary rebuilt and installed: `go install ./cmd/govalid/`

### Final Verification
- [ ] All GitHub Actions CI checks pass
- [ ] No breaking changes to existing functionality
- [ ] Contribution follows project license requirements

### CODE REVIEW

File: internal/analyzers/markers/markers.go [hunk 1]
```diff
@@ -42,17 +55,15 @@ func (m *markers) FieldMarkers(field *ast.Field) MarkerSet {
 	return m.fieldMarkers[field]
 }
 
-// insertFieldMarkers adds a set of markers to a specific struct field.
-func (m *markers) insertFieldMarkers(field *ast.Field, markers MarkerSet) {
-	if len(markers) == 0 {
-		return
-	}
-
+// insertFieldMarkers adds a marker to a specific struct field.
```
Comment (t4kamura) [inline internal/analyzers/markers/markers.go:58]:
```suggestion
// insertFieldMarker adds a marker to a specific struct field.
```

**Review by t4kamura — CHANGES_REQUESTED** [review]
@sivchari 
The method names in godoc remained in the previous naming.
Other than that, no issues, I think it's a good change!

File: internal/analyzers/markers/markers.go (hunk 1 above)
Comment (sivchari) [reply internal/analyzers/markers/markers.go:59]: Thanks!
